py/selenium/webdriver/common/bidi/browser.py

```diff
@@ -123,16 +123,51 @@ def from_dict(cls, data: dict) -> "ClientWindowInfo":
         Returns:
         -------
             ClientWindowInfo: A new instance of ClientWindowInfo.
+
+        Raises:
+        ------
+            ValueError: If required fields are missing or have invalid types.
         """
-        return cls(
-            client_window=data.get("clientWindow"),
-            state=data.get("state"),
-            width=data.get("width"),
-            height=data.get("height"),
-            x=data.get("x"),
-            y=data.get("y"),
-            active=data.get("active"),
-        )
+        try:
+            client_window = data.get("clientWindow")
+            if not isinstance(client_window, str):
+                raise ValueError("clientWindow must be a string")
+
+            state = data.get("state")
+            if not isinstance(state, str):
+                raise ValueError("state must be a string")
+
+            width = data.get("width")
+            if not isinstance(width, int):
+                raise ValueError("width must be an integer")
+
+            height = data.get("height")
+            if not isinstance(height, int):
+                raise ValueError("height must be an integer")
+
+            x = data.get("x")
+            if not isinstance(x, int):
+                raise ValueError("x must be an integer")
```

Comment:
```suggestion
			x = data["x"]
            if not isinstance(x, int):
                raise ValueError(f"x must be an integer, got {type(x).__name__}")
```
x and y can be int which is correct.